collect.py

```diff
@@ -231,49 +256,72 @@ def get_screenshot_and_domtree(driver, bug_id, browser, seq_no=None):
     get_domtree(driver, bug_id, browser, seq_no)
 
 
-def run_test(bug, browser, driver, op_sequence=None):
-    print('Testing %s (bug %d) in %s' % (bug['url'], bug['id'], browser))
+def count_lines(bug_id):
+    try:
+        with open('data/%d.txt' % bug_id) as f:
+            return sum(1 for line in f)
+    except IOError:
+        return 0
+
+
+# We restart from the start and follow the saved sequence
+def jump_back(current_path, firefox_driver, chrome_driver, visited_paths, bug):
+    firefox_driver.get(bug['url'])
+    chrome_driver.get(bug['url'])
+    for elem_properties in current_path:
+        do_something(firefox_driver, visited_paths, current_path, elem_properties)
+        do_something(chrome_driver, visited_paths, current_path, elem_properties)
+
+
+def run_test_both(bug, firefox_driver, chrome_driver):
+    print('Testing %s (bug %d) in %s' % (bug['url'], bug['id'], 'firefox'))
 
     try:
-        driver.get(bug['url'])
+        firefox_driver.get(bug['url'])
     except TimeoutException as e:
         # Ignore timeouts, as they are too frequent.
         traceback.print_exc()
         print('Continuing...')
 
-    get_screenshot_and_domtree(driver, str(bug['id']), browser)
+    get_screenshot_and_domtree(firefox_driver, str(bug['id']), 'firefox')
+    print('Testing %s (bug %d) in %s' % (bug['url'], bug['id'], 'chrome'))
 
-    saved_sequence = []
     try:
-        max_iter = 7 if op_sequence is None else len(op_sequence)
-        for i in range(0, max_iter):
-            if op_sequence is None:
-                elem_properties = do_something(driver)
-                if elem_properties is None:
-                    print('Can\'t find any element to interact with on %s for bug %d' % (bug['url'], bug['id']))
-                    break
-                saved_sequence.append(elem_properties)
-            else:
-                elem_properties = op_sequence[i]
-                do_something(driver, elem_properties)
-
-            print('  - Using %s' % elem_properties)
-            get_screenshot_and_domtree(driver, str(bug['id']), browser, str(i))
-
+        chrome_driver.get(bug['url'])
     except TimeoutException as e:
         # Ignore timeouts, as they are too frequent.
         traceback.print_exc()
         print('Continuing...')
 
-    return saved_sequence
+    get_screenshot_and_domtree(chrome_driver, str(bug['id']), 'chrome')
 
+    visited_paths = []
+    current_path = []
+    while True:
+        elem_properties = do_something(firefox_driver, visited_paths, current_path)
+        print(elem_properties)
```

Comment:
Print as before, to have more verbose output: `print('  - Using %s' % elem_properties)`